tests/Commit/BatchCommitterTest.php

```diff
@@ -14,14 +15,18 @@ public function testShouldCommitMessageOnlyAfterTheBatchSizeIsReached()
         $committer = $this->createMock(Committer::class);
         $committer
             ->expects($this->exactly(2))
-            ->method('commitMessage');
+            ->method('commitMessage')
+            ->withConsecutive(
+                [$this->isInstanceOf(Message::class), true],
+                [$this->isInstanceOf(Message::class), true]
+            );
 
         $batchSize = 3;
         $messageCounter = new MessageCounter(42);
         $batchCommitter = new BatchCommitter($committer, $messageCounter, $batchSize);
 
         for ($i = 0; $i < 7; $i++) {
-            $batchCommitter->commitMessage();
+            $batchCommitter->commitMessage(new Message(), true);
```

Comment:
Can `success` be true by default on `commitMessage` method, then we only pass this argument when false?

Reply:
Sure it can, though I'm not fully convinced that is the best design for the interface itself.

The error handling of the lib has to be solid and will be relied upon by users/dependents, and I believe that always being explicit about an input value reduces the amount of assumptions in the code (a default value = assumption).

Actually, in practice the default value is not used at all since all calls to `commitMessage()` have the `$success` variable available, so it is always set explicitly.

The default value could be used in the unit tests, but we can just fill the `$success` param there with `true`.

What do you think?

Reply:
Ok, let's keep it without default value for now.